docs/latest/tutorial/ipc.md

```diff
@@ -0,0 +1,554 @@
+---
+title: Inter-Process Communication
+description: Use the ipcMain and ipcRenderer modules to communicate between Electron processes
+slug: ipc
+hide_title: false
+---
+
+# Inter-Process Communication
+
+Inter-process communication (IPC) is key part of building feature-rich desktop applications
+in Electron. Because the main and renderer processes have different responsibilities in
+Electron's process model, IPC is the only way to perform many common tasks, such as calling a
+native API from your UI or triggering changes in your web contents from native menus.
+
+## IPC channels
+
+In Electron, processes communicate by passing messages through developer-defined "channels"
+with the [`ipcMain`][ipc-main] and [`ipcRenderer`][ipc-renderer] modules. These channels are
+**arbitrary** (you can name them anything you want) and **bidirectional** (you can use the
+same channel name for both modules).
+
+In this guide, we'll be going over some fundamental IPC patterns with concrete examples that
+you can use as a reference for your app code.
+
+[ipc-main]: ../api/ipc-main.md
+[ipc-renderer]: ../api/ipc-renderer.md
+
+## Understanding context-isolated processes
+
+Before proceeding to implementation details, you should be familiar with the idea of using a
+[preload script] to import Node.js and Electron modules in a context-isolated renderer process.
+
+* For a full overview of Electron's process model, you can read the [process model docs].
+* For a primer into exposing APIs from your preload script using the `contextBridge` module, check
+out the [context isolation tutorial].
+
+[preload script]: process-model.md#preload-scripts
+[process model docs]: process-model.md
+[context isolation tutorial]: context-isolation.md
```

Comment:
Once the guide is OK can we move all the label links at the bottom of the doc?
I've had some issues when refactoring the docs in the tutorials and updating to new references.